### DISCUSSION TITLE
Why do we use 'use server' in the server action component?

### DISCUSSION THREAD

**Discussion by kemaltf** [vercel/next.js#58752] [Help] [answered]
### Summary

Hi all, 

I just read the [NextJS 14 tutorial](https://nextjs.org/learn/dashboard-app/mutating-data), and found this:

 
```
// /app/lib/actions.ts

'use server';
 
export async function createInvoice(formData: FormData) {
  const rawFormData = {
    customerId: formData.get('customerId'),
    amount: formData.get('amount'),
    status: formData.get('status'),
  };
  // Test it out:
  console.log(rawFormData);
}
```

the tutorial said that '_**By adding the 'use server', you mark all the exported functions within the file as server functions.**_' 

**and i don't get it the point.**

I mean in next js 14, the component is already 'server' side by default, so why we need to add `use server` on top of that?
And also, when i try to delete the `use server` it comes error. 

Any explanation why we add use server when we use server action in next js? thanks

### Additional information

<img width="701" alt="image" src="https://github.com/vercel/next.js/assets/91187830/8c0ade2a-384d-4213-9255-de3eb62b2213">

**Comment by icyJoseph** [comment]
From, https://react.dev/reference/react/use-server#noun-labs-1201738-(2):

> 'use server' marks server-side functions that can be called from client-side code.

**Reply by kemaltf** [reply]
but what i undrestand is by default all the component and page is on server side right?

**Reply by icyJoseph** [reply, marked as answer]
Right, React Server Components do not send their JavaScript to the client, at module level.

However, if you were to import one of these Action functions into a Client Component module, then you'd send their JavaScript to the client, and whole point with Server Actions is to not do that. Note that the definition clearly states:

> server-side functions that can be called from client-side code

That's the key aspect of slapping `use server` at module level, and even within the body of a function:

```ts
async function foo() {
 'use server';
 // rest of the function
}
```

**Reply by kemaltf** [reply]
Thanks! after learn about server component in React, now finally i can understand the purpose of making "use server" inside the client component. 💯
